integration-tests/src/test/java/eu/tsystems/mms/tic/testframework/test/webdrivermanager/DesktopWebDriverFactoryTest.java

```diff
@@ -21,63 +21,89 @@
         request.setBaseUrl("http://google.de");
-        //request.webDriverMode = WebDriverMode.local;
-        //request.browser = Browsers.phantomjs;
 
-        WebDriver driver = WebDriverManager.getWebDriver(request);
+        WebDriver driver = WEB_DRIVER_MANAGER.getWebDriver(request);
         String currentUrl = driver.getCurrentUrl();
         Assert.assertTrue(currentUrl.contains("google"), "Current URL contains google - actual: " + currentUrl);
     }
 
     @Test
-    public void testT02_BaseURL_NotSet() throws Exception {
+    public void testT02_BaseURL_NotSet() {
         DesktopWebDriverRequest request = new DesktopWebDriverRequest();
-        //request.webDriverMode = WebDriverMode.local;
-        //request.browser = Browsers.phantomjs;
 
-        WebDriver driver = WebDriverManager.getWebDriver(request);
+        WebDriver driver = WEB_DRIVER_MANAGER.getWebDriver(request);
         String currentUrl = driver.getCurrentUrl();
         // Empty baseUrl of Chrome
         Assert.assertTrue(currentUrl.contains("data"), "Current URL is invalid - actual: " + currentUrl);
     }
 
     @Test
-    public void testT03_EndPointCapabilities() throws Exception {
+    public void testT03_EndPointCapabilities_WebDriverRequest() throws Exception {
         DesktopWebDriverRequest request = new DesktopWebDriverRequest();
         request.setBaseUrl("http://google.de");
-        //request.webDriverMode = WebDriverMode.local;
-        //request.browser = Browsers.phantomjs;
 
-        /*
-        create caps
-         */
-        DesiredCapabilities caps = new DesiredCapabilities();
+        DesiredCapabilities caps = request.getDesiredCapabilities();
         caps.setCapability("enableVideo", true);
         caps.setCapability("enableVNC", true);
 
-        // start session
-        WebDriver driver = WebDriverManager.getWebDriver(request);
-        SessionContext sessionContext = WebDriverSessionsManager.getSessionContext(driver).get();
+        WebDriver driver = WEB_DRIVER_MANAGER.getWebDriver(request);
+
+        SessionContext sessionContext = WEB_DRIVER_MANAGER.getSessionContext(driver).get();
+        Map<String, Object> sessionCapabilities = sessionContext.getWebDriverRequest().getCapabilities();
+
+        Assert.assertEquals(sessionCapabilities.get("enableVideo"), caps.getCapability("enableVideo"), "EndPoint Capability via WebDriverRequest is set");
+        Assert.assertEquals(sessionCapabilities.get("enableVNC"), caps.getCapability("enableVNC"), "EndPoint Capability via WebDriverRequest is set");
+    }
+
+    @Test
+    public void testT04_EndPointCapabilities_Global() {
+        WEB_DRIVER_MANAGER.setGlobalCapability("t04Global", "yes");
+
+        WebDriver driver = WEB_DRIVER_MANAGER.getWebDriver();
+
+        WEB_DRIVER_MANAGER.removeGlobalCapability("t04Global");
+
+        SessionContext sessionContext = WEB_DRIVER_MANAGER.getSessionContext(driver).get();
```

Comment:
`@SuppressWarnings("OptionalGetWithoutIsPresent")`?

Reply:
There _must_ be a session context, otherwise the test will fail.

Reply:
Yes, thence I suggest to suppress the warning. It helps keeping warnings relevant. ... Feel free to skip though - I did not check for this thoroughly, just noticed because I had the file open anyway.